addons/mail/static/src/model/model_manager.js

```diff
@@ -708,6 +710,16 @@ class ModelManager {
         for (const record of this._updatedRecords) {
             record.__state++;
         }
+
+        // handle required field: validate that mandatory are set.
+        for (const record of this._updatedRecords) {
+            for (const required of record.constructor.__requiredFieldsList) {
+                if (record[required] === undefined) {
+                    throw Error(`Field ${required.fieldName} of ${record.localId} is required.`);
```

Comment:
Note that in case of unmet required fields that are used to compute local id, would not be straight-forward where the issue comes. Also another crash may occur somewhere else, like _"A record already exists for model [...] with localId [...]"_.

Not worth improving this right now. Just wanted to point out limitation(s) of required field check of this PR.